### PR TITLE
Add Vagrantfile for VM Provisioning and update README

### LINKED ISSUE #14: Deploy to a "cloud" server
* Deploy the current release of your _ITU-MiniTwit_, see task 2 b), to a remote ("cloud") server.
    - You might want to deploy to DigitalOcean, from where you get free credits from the GitHub education pack.
    - You might want to start with the exercise from the demo in the lecture, see slide [g) Automatic provision of remote clusters of virtual machines at DigitalOcean via Vagrant](./Slides.md#g-Automatic-provision-of-remote-clusters-of-virtual-machines-at-DigitalOcean-via-Vagrant).
    - **OBS**: The simulator API from the task above is considered part of your _ITU-MiniTwit_, i.e., it has to be deployed too.
  * **Do not** create and provision VMs manually, instead use a corresponding script, `Vagrantfile`, or a similar reproducible setup for that task.

  * You are free to deploy to whichever Infrastructure-as-a-Service (IaaS) provider of your choice or your own servers.
    - The only requirement is that your deployment is automatically reproducible, i.e., no clicking in UIs required, and that your application is reachable via a public IP/domain.
    - **Do not** deploy to a Platform-as-a-Service, like Heroku, Google App-Engine, etc.

  * Make your development environment and deployment environment "look alike". That is, you develop on an Ubuntu 22.04 based Linux (Pop!_OS), choose the same environment for your remote server(s).

  * **Hint**: To make future operations more smooth, you might consider to use a [floating IP](https://docs.digitalocean.com/products/networking/floating-ips/) 'infront' of your actual server, see the [API spec.](https://docs.digitalocean.com/reference/api/api-reference/#tag/Floating-IPs) for receiving one.

### PR DESCRIPTION
**Pull request by duckth**
Closes #14

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  config.vm.define "droplet1" do |droplet|
+    droplet.vm.provider :digital_ocean do |provider, override|
+      override.ssh.private_key_path = PRIVATE_KEY_PATH
+      override.vm.box = 'digital_ocean'
+      override.vm.box_url = DO_BOX_URL
+      override.nfs.functional = false
+      override.vm.allowed_synced_folder_types = :rsync
+      provider.token = TOKEN
+      provider.image = 'ubuntu-22-04-x64'
+      provider.region = 'nyc1'
```
Comment (fredpetersen) [inline Vagrantfile:16]: New york city er OK langt væk
```suggestion
      provider.region = 'fra1'
```

**Comment by duckth** [comment]
Need to fix synced folder

**Comment by duckth** [comment]
http://165.227.245.161:5001

**Comment by silkeholmebonnen** [comment]
There are 3 different url mentioned in the PR; http://165.227.245.161:5001/ http://134.209.131.46:5000/ and http://67.205.156.216:5000/public but none of them works for me. On the first one I get an error message saying "You are not authorized to use this resource!" and the other ones just don't work. Which one do we use and is it supposed to work at the moment?

**Comment by duckth** [comment]
> There are 3 different url mentioned in the PR; http://165.227.245.161:5001/ http://134.209.131.46:5000/ and http://67.205.156.216:5000/public but none of them works for me. On the first one I get an error message saying "You are not authorized to use this resource!" and the other ones just don't work. Which one do we use and is it supposed to work at the moment?

Apologies for the terrible communication about this:
http://165.227.245.161:5001/ (port 5001) is for the simulator API only. You need the credentails from the simulator to access it
http://165.227.245.161:5000/ (port 5000) is the web interface

The other URL's are outdated.

**Comment by duckth** [comment]
I'll update the readme at some point

**Review by fredpetersen — APPROVED** [review]

**Review by fredpetersen — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -33,3 +33,16 @@ docker run --rm -v /tmp:/tmp -it -p 5000:5000 minitwit [cmd]
...
 
+
+### VM Provisioning & Deployment 
+
+Virtual Machine provisioning is handled by Vagrant in DigitalOcean.
+Run provisioning with `vagrant up`
+The environment variables `DIGITAL_OCEAN_PRIVATE_KEY_PATH`, `SSH_KEY_NAME` and `DIGITAL_OCEAN_PAT` are needed.
+The first is the path to the authenticated private key, the second is the name of your SSH key for Digital Ocean and the third is the authenticated token for the team
+
+The Vagrantfile syncs the local repository to the VM so be sure to have a database in `db/minitwit.db`
```
Comment (A-Guldborg) [inline README.md:44]:
```suggestion
The Vagrantfile syncs the local repository to the VM so be sure to have a database in `./db/minitwit.db`
```

**Review by A-Guldborg — CHANGES_REQUESTED** [review]

**Review by A-Guldborg — APPROVED** [review]
LGTM

**Pull request merged**